### ISSUE TITLE
Better way to handle transactions e.g. via arangojs

### ISSUE THREAD

**Issue by maku** [arangodb/arangodb#4780]
For me the possibility to have transaction support is a huge selling point for ArangoDB. 
I am using arangojs via node. So I played around with transaction support and it feels rather painfull how it works (especially with more complex functions) e.g. my code is written in typescript now I have to write a javascript function as a string.

So my question: would it be possible to provide an API e.g. like firestore (https://firebase.google.com/docs/firestore/manage-data/transactions) to handle transactions?

I know it would be possible to use foxx. But from my point of view usage of foxx further enhances the complexity of the build / testing/ tooling etc. (coding in typescript, etc) especially when the most part of data base logic is already coded in a simple nodejs backend.

**Comment by oliverhausler** [comment]
I agree with @maku. Using foxx is a workaround. RocksDB apparently supports transactions natively, so it should be possible to implement this feature to be used from JS/AQL directly.

https://www.bookstack.cn/read/rocksdb-en/b603e47dd8805bbf.md

**Comment by dothebart** [comment]
Meanwhile [Stream Transactions](https://docs.arango.ai/arangodb/stable/develop/transactions/stream-transactions/) are available and portable across all drivers and targets without having to code javascript for the transaction itself.

Closing.
